pkg/arvo/sys/vane/ames.hoon

```diff
@@ -3061,14 +3061,20 @@
   (en:crub:crypto symmetric-key (jam plaintext))
 ::  +decrypt: decrypt packet content to a $shut-packet or die
 ::
+::    TODO: alter |acru interface
+::
 ++  decrypt
   |=  [=symmetric-key ciphertext=@]
   ^-  shut-packet
-  ::
-  ;;  shut-packet
-  %-  cue
-  %-  need
-  (de:crub:crypto symmetric-key ciphertext)
+  =+  ;;([iv=@ len=@ cyf=@] (cue ciphertext))
+  ::  assert cyphertext length <= claimed length <= max bytes
+  ::
+  ?.  (lte (met 3 cyf) len)
+    ~|(ames-len-small+len^(met 3 cyf) !!)
```

Comment:
This first assertion is actually redundant, the AES-CTR implementations already include it:

https://github.com/urbit/urbit/blob/ca3872fba46660e3638b377cdf4bd0f64b535a7f/pkg/arvo/sys/zuse.hoon#L3021

The absence of the same in the AES-SIV jet is a mismatch (as is the length truncation to `c3_w`).

Reply:
Ah, ok so it sounds like this first assertion should be removed from Ames, relying instead on Zuse's crypto code to enforce this.  That sounds ok.  I'll remove this, then.

Reply:
To be clear, the length assertion is not reflected in the AES-SIV jet, which is an active mismatch.